File: 492. Construct the Rectangle/solution.go
func constructRectangle(area int) []int {
	var ret []int
	for W := 1; W*W <= area; W++ {
		if area%W == 0 {
			L := area / W

			if W <= L {
				ret = []int{L, W}
			} else {
				return ret
			}
		}
	}

	return ret
}

File: 492. Construct the Rectangle/solution.js
/**
 * @param {number} area
 * @return {number[]}
 */
var constructRectangle = function(area) {
  let w = Math.trunc(Math.sqrt(area));
  while (area % w !== 0) w -= 1;
  return [area / w, w];
};
